Tests/GettableSpec.swift

```diff
@@ -29,5 +52,43 @@ class GettableSpec: QuickSpec {
             
             itBehavesLike("a Restable") { context }
         }
+        
+        describe("A gettable with a dateDecodingStrategy") {            
+            it("It decodes the response data according to the dateDecodingStrategy") {
+                let gettable = DateRequest()
+                let currentDateString = "2018-06-12T20:00:00.000-04:00"
+                let currentDate = gettable.dateFormatter().date(from: currentDateString)!
+                
+                do {
+                    // encode the date
+                    let encoder = JSONEncoder()
+                    encoder.dateEncodingStrategy = .formatted(gettable.dateFormatter())
+                    let data = try encoder.encode(["date": currentDateString])
+                    var dateResponse: DateResponse? = nil
+                    
+                    // NOTE: We use a dummy url response here as the `dataTaskCompletionHandler` needs it.
+                    // As long as this method gets a success statusCode (ie. 200) from an HTTPURLResponse, it will proceed to decode the data passed in.
+                    // We want to verify that it is decoding it according to the `dateDecodingStrategy` of the `DateRequest`
+                    let response = HTTPURLResponse(url: URL(string: "http://google.ca")!, statusCode: 200, httpVersion: nil, headerFields: nil)
+                    gettable.dataTaskCompletionHandler(data: data, response: response, error: nil) { response in
+                        if case let .success(tempDateResponse) = response {
+                            dateResponse = tempDateResponse
+                        }
+                    }
+                    
+                    // compare the dates
+                    let currentComponents = Calendar.current.dateComponents([.year, .month, .day, .hour, .minute, .second], from: currentDate)
+                    var actualComponents: DateComponents? = nil
+                    if let dateResponse = dateResponse {
```

Comment:
instead of using an `if-let` here, it would be better, likely on line `79` before we do anything else, to ensure dateResponse isn't nil, and fail the test if it doesn't.

```Swift
expect(dateResponse).toNot(beNil())
```

After which just force-unwrap wherever you need it.

Reply:
👍